src/pages/AdminPage.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addProduct, removeProduct } from "../store/products/actions";
+import { selectAllProducts } from "../store/products/selectors";
+import ProductCard from "../components/ProductCard";
+import "./AdminPage.scss";
+
+export default function AdminPage() {
+  const dispatch = useDispatch();
+  const [form, setForm] = useState({
+    name: "",
+    description: "",
+    url: "",
+    price: 0,
+    tags: [],
+    categories: [],
+  });
+  const [idToRemove, setIdToRemove] = useState("");
+  const products = useSelector(selectAllProducts);
+
+  const submitAddProductForm = (event) => {
+    event.preventDefault();
+
+    dispatch(
+      addProduct(
+        form.name,
+        form.description,
+        form.url,
+        Number(form.price),
+        form.tags,
+        form.categories
+      )
+    );
+
+    setForm({
+      name: "",
+      description: "",
+      url: "",
+      price: 0,
+      tags: [],
+      categories: [],
```

Comment:
Small DRY comment: if you define this outside the component like:
```
const formInitialState = {
    name: "",
    description: "",
    url: "",
    price: 0,
    tags: [],
    categories: [],
  };
```
Then you can do:
```
const [form, setForm] = useState(formInitialState)

// and later....

setForm(formInitialState)
```
Consider this for big object states